### PR TITLE
[stable10] Move AccountCheckException to public api

### PR DESCRIPTION
**Pull request by DeepDiver1975**
backport #31915

### CODE REVIEW

File: lib/private/User/Session.php [hunk 1]
```diff
@@ -517,7 +518,6 @@ private function loginWithPassword($login, $password) {
 
 		$user = $this->manager->checkPassword($login, $password);
 		if ($user === false) {
-			$this->manager->emit('\OC\User', 'failedLogin', [$login]);
```
Comment (PVince81) [inline lib/private/User/Session.php]: what happened to these ? obsolete ?

File: lib/private/User/Session.php [hunk 2]
```diff
@@ -463,6 +463,7 @@ public function prepareUserLogin($firstTimeLogin = false) {
 
 			// trigger any other initialization
 			$this->eventDispatcher->dispatch(IUser::class . '::firstLogin', new GenericEvent($this->getUser()));
+			$this->eventDispatcher->dispatch('user.firstlogin', new GenericEvent($this->getUser()));
```
Comment (PVince81) [inline lib/private/User/Session.php]: why needing another event ?

Comment (DeepDiver1975) [reply lib/private/User/Session.php]: to fix the naming convention

File: lib/private/User/Session.php (hunk 1 above)
Comment (DeepDiver1975) [reply lib/private/User/Session.php]: the emit call is within emitFailedLogin

File: lib/private/User/Session.php [hunk 3]
```diff
@@ -1168,11 +1170,13 @@ protected function getAuthModules($includeBuiltIn) {
...
 
 	/**
-	 * This method triggers symfony event for failed login
-	 *
+	 * This method triggers symfony event for failed login as well as
+	 * emits via the emitter in user manager
 	 * @param string $user
 	 */
 	protected function emitFailedLogin($user) {
+		$this->manager->emit('\OC\User', 'failedLogin', [$user]);
```
Comment (DeepDiver1975) [inline lib/private/User/Session.php]: @PVince81 see here

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**